src/main/java/org/cloudfoundry/promregator/scanner/ReactiveAppInstanceScanner.java

```diff
@@ -306,12 +329,33 @@ public void stop() {
 				tuple.getT2().stop();
 				return tuple.getT1();
 			}).cache();
-			
+
 			this.applicationMap.put(key, cached);
 			return cached;
 		}
 	}
-	
+
+	private Flux<ApplicationResource> allApplicationByOrgAndSpace(Mono<String> orgIdMono, Mono<String> spaceIdMono) {
+
+		return Flux.zip(orgIdMono, spaceIdMono, Mono.just(PromregatorApplication.SPACE_ALL_APPLICATIONS))
+				.flatMap(triple ->
+						this.cfAccessor.retrieveApplicationId(triple.getT1(), triple.getT2(), triple.getT3())
+				).flatMap(response -> {
+					List<ApplicationResource> resources = response.getResources();
+					if (resources == null) {
+						return Flux.empty();
+					}
+
+					if (resources.isEmpty()) {
+						log.warn(String.format("Received empty result on requesting application %s",
+								PromregatorApplication.SPACE_ALL_APPLICATIONS));
+						return Flux.empty();
+					}
+
+					return Flux.fromIterable(resources);
+				}).filter(r -> r.getEntity().getState().equalsIgnoreCase("STARTED"));
```

Comment:
That's an important aspect that we still need to cover in our discussion at #35 :  Apps in which state do we consider that we should discover for scraping? You have filtered on `STARTED` here. What about `STOPPED`? Are there further "bad states" (`CRASHED`?)?

BTW: Aren't there any constants interfaces for these literals?

Reply:
This relevant indeed. For my prototype i've pre-filtered only the STARTED apps but i  don't fill like this is correct generic behavior. We can either the state an yet another configurable filter or register all targets in spite of their sates but pass the state as dimensional tag along the measurements.  Not entirely sure how would this work with Prometheus though,.

Reply:
I think we have to iterate on that a little. `STARTED` is correct for sure, but it most likely won't be the only one. Yet, `CRASHED` or `STOPPED` will not make sense to let through, as then also the scraping endpoint won't be available anyway. Even if it was, most likely nothing meaningful could come back. So, I would assume that Prometheus would want to see those instances with an
``` 
up{...} 0
```
anyway.
Making it configurable (in the special event that really some folks have deviating opinion) can also be done compatibly afterwards.